test/distributed/org/apache/cassandra/distributed/impl/MessageFilters.java

```diff
@@ -19,28 +19,23 @@
 package org.apache.cassandra.distributed.impl;
 
 import java.util.Arrays;
-import java.util.Set;
-import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
 
-import org.apache.cassandra.distributed.api.IInstance;
+import org.apache.cassandra.distributed.api.IMessage;
 import org.apache.cassandra.distributed.api.IMessageFilters;
-import org.apache.cassandra.net.Verb;
 
 public class MessageFilters implements IMessageFilters
 {
-    private final Set<Filter> filters = new CopyOnWriteArraySet<>();
+    private final List<Filter> filters = new CopyOnWriteArrayList<>();
 
-    public boolean permit(IInstance from, IInstance to, int verb)
+    public boolean permit(int from, int to, IMessage msg)
     {
-        if (from == null || to == null)
-            return false; // cannot deliver
-        int fromNum = from.config().num();
-        int toNum = to.config().num();
-
         for (Filter filter : filters)
-            if (filter.matches(fromNum, toNum, verb))
-                return false;
-
+        {
+            if (filter.matches(from, to, msg))
+                return filter.permitIfMatched;
```

Comment:
this feels weird to me; if I have the following

```
filter(verb).pass();
filter(verb).drop();
```

what happens?  It looks like the drop will be skipped in favor of pass.  Don't see a test for this since each test calls `filters.reset();` before adding the second filter.

The pass/drop stuff already exists so nothing for this PR, but I feel that `pass()` should be removed (should file jira for this)

Reply:
+1

Reply:
We also can achieve the same thing by simply returning `false` and not matching on custom matcher.